src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs

```diff
@@ -80,11 +81,18 @@ public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel,
             foreach (var model in context.Library.ArmResource)
             {
                 var codeWriter = new CodeWriter();
-                armResourceWriter.WriteResource(codeWriter, model);
+                armResourceWriter.WriteResource(codeWriter, model.Value);
 
-                var name = model.Type.Name;
+                var name = model.Value.Type.Name;
                 project.AddGeneratedFile($"{name}.cs", codeWriter.ToString());
             }
+
+            var extensionsWriter = new CodeWriter();
+            var resources = context.Library.ArmResource;
+            var mgmtConfiguration = context.Configuration.MgmtConfiguration;
+            var @namespace = context.DefaultNamespace;
+            resourceGroupExtensionsWriter.WriteExtension(@namespace, mgmtConfiguration, extensionsWriter, resources);
```

Comment:
Can we just pass context in here instead?

Reply:
Done